### PR TITLE
Changed test-task to not enable stateChangesOnly

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

https://jira.rax.io/browse/SALUS-947

# What

This a follow up to #53 to ensure that sequential, non-OK states are each emitted by kapacitor and therefore included in final response.

# How

- Conditionalized the `stateChangesOnly` designation
- While I was in the code, noticed that the `tenantId` didn't actually need to be passed to the `TickScriptBuilder` since tenant gets "encoded" into the task ID

# How to test

Updated unit tests

# Example

Request:
```json
{
	"task": {
		"measurement": "mem",
		"taskParameters": {
			"criticalStateDuration": 2,
			"warningStateDuration": 2,
			"infoStateDuration": 2,
			"stateExpressions": [
				{
					"state": "WARNING",
					"expression": {
						"type": "comparison",
						"comparator": "<",
						"metricName": "free",
						"comparisonValue": 3000
					}
				},
        {
					"state": "CRITICAL",
					"expression": {
						"type": "comparison",
						"comparator": "<",
						"metricName": "free",
						"comparisonValue": 1000
					}
				}
			]
		}
	},
	"metrics": [
    {
			"name": "mem",
			"ivalues": {
				"free": 2500
			}
		},
    {
			"name": "mem",
			"ivalues": {
				"free": 2000
			}
		},
    {
			"name": "mem",
			"ivalues": {
				"free": 500
			}
		},
    {
			"name": "mem",
			"ivalues": {
				"free": 3500
			}
		}
	]
}
```

response:
```
{
  "partialResults": false,
  "events": [
    {
      "level": "WARNING",
      "data": {
        "series": [
          {
            "name": "mem",
            "columns": [
              "time",
              "free"
            ],
            "values": [
              [
                "2020-07-23T19:05:54.4370717Z",
                2500
              ]
            ]
          }
        ]
      }
    },
    {
      "level": "WARNING",
      "data": {
        "series": [
          {
            "name": "mem",
            "columns": [
              "time",
              "free"
            ],
            "values": [
              [
                "2020-07-23T19:05:54.4490173Z",
                2000
              ]
            ]
          }
        ]
      }
    },
    {
      "level": "CRITICAL",
      "data": {
        "series": [
          {
            "name": "mem",
            "columns": [
              "time",
              "free"
            ],
            "values": [
              [
                "2020-07-23T19:05:54.4590482Z",
                500
              ]
            ]
          }
        ]
      }
    },
    {
      "level": "OK",
      "data": {
        "series": [
          {
            "name": "mem",
            "columns": [
              "time",
              "free"
            ],
            "values": [
              [
                "2020-07-23T19:05:54.4643862Z",
                3500
              ]
            ]
          }
        ]
      }
    }
  ],
  "stats": {
    "task-stats": {
      "throughput": 0
    },
    "node-stats": {
      "alert2": {
        "alerts_inhibited": 0,
        "alerts_triggered": 4,
        "avg_exec_time_ns": 0,
        "collected": 4,
        "crits_triggered": 1,
        "emitted": 0,
        "errors": 0,
        "infos_triggered": 0,
        "oks_triggered": 1,
        "warns_triggered": 2,
        "working_cardinality": 1
      },
      "from1": {
        "avg_exec_time_ns": 0,
        "collected": 4,
        "emitted": 4,
        "errors": 0,
        "working_cardinality": 0
      },
      "stream0": {
        "avg_exec_time_ns": 0,
        "collected": 4,
        "emitted": 4,
        "errors": 0,
        "working_cardinality": 0
      }
    }
  }
}
```

### CODE REVIEW

File: src/main/java/com/rackspace/salus/event/manage/services/TasksService.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public EventEngineTask createTask(String tenantId, CreateTask in) {
             .setDb(tenantId)
             .setRp(InfluxScope.INGEST_RETENTION_POLICY)
             ))
-        .setScript(tickScriptBuilder.build(tenantId, in.getMeasurement(), in.getTaskParameters()))
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/event/manage/services/TasksService.java:83]: Thanks for cleaning up all this stuff.  I had noticed it when first looking into event-engine.

**Comment by jjbuchan** [comment]
Noticed tests are failing for this.  Maybe it just needs a rebase?

**Comment by itzg** [comment]
> Noticed tests are failing for this. Maybe it just needs a rebase?

Yeah, got them all fixed at one point, so was circling back to that now.

**Comment by itzg** [comment]
I haven't noticed Mockito being this picky before, but something about the overloaded method required more specific argument matchers:

https://github.com/racker/salus-event-engine-management/pull/54/commits/4e7c0f590cf15a61ba846c72f77ce2f25795ed92#diff-444933bb39959d764bcbdfd8d7b14e0dR194

**Comment by jjbuchan** [comment]
> I haven't noticed Mockito being this picky before

I obviously didn't review the tests closely enough.  I've hit that problem numerous times.  I think even for non-overloaded methods I still specify the specific `anyX`.

**Comment by itzg** [comment]
> I've hit that problem numerous times. I think even for non-overloaded methods I still specify the specific `anyX`.

Apparently I have been lulled into thinking I could be lazy with `any()`'s everywhere :)

**Pull request merged**